lightning: refine some TiDB Lightning docs (#13779)
pingcap/docs: migrate-aurora-to-tidb.md

## Prerequisites

> - [Install Dumpling and TiDB Lightning](/migration-tools.md). If you want to create the corresponding tables manually on the target side, do not install Dumpling.

Author: ```suggestion
- [Install Dumpling and TiDB Lightning](/migration-tools.md). If you want to create the corresponding tables manually on the target side, do not install Dumpling.
- [Get the upstream database privileges required by Dumpling](/dumpling-overview.md#required-privileges).
```

> - [Get the target database privileges required for TiDB Lightning](/tidb-lightning/tidb-lightning-faq.md#what-are-the-privilege-requirements-for-the-target-database).

Reviewer: 英文的比中文少了一条`- [获取 Dumpling 所需上游数据库权限](/dumpling-overview.md#需要的权限)`，这是特意移除的吗？

Author: 应该是漏掉了，加上了。

## Import full data to TiDB

### Step 1. Export and import the schema file

This section describes how to export the schema file from Amazon Aurora and import it to TiDB. If you have manually created the table in the target database, you can skip this step.

#### 1.1 Export the schema file from Amazon Aurora

Because the snapshot file from Amazon Aurora does not contain the DDL statements, you need to export the schema using Dumpling and create the schema in the target database using TiDB Lightning.

> Export the schema using Dumpling by running the following command. The command includes the `--filter` parameter to only export the desired table schema. For more information about the parameters, see [Option list of Dumpling](/dumpling-overview.md#option-list-of-dumpling).

Reviewer: ```suggestion
Export the schema using Dumpling by running the following command. The command includes the `--filter` parameter to only export the desired table schema. For more information about the parameters, see the [Option list of Dumpling](/dumpling-overview.md#option-list-of-dumpling).
```

```shell
export AWS_ACCESS_KEY_ID=${access_key}
export AWS_SECRET_ACCESS_KEY=${secret_key}

> tiup dumpling --host ${host} --port 3306 --user root --password ${password} --filter 'my_db1.table[12]' --no-data --output 's3://my-bucket/schema-backup' --filter "mydb.*"

Reviewer: - 中英文的命令不完全一样，中文的 filter 两条规则是合在一起的，英文是分开的？建议统一用一种写法
- 英文里没有 `--consistency none`

Reviewer 2: @okJiang 帮忙确认一下，谢谢！

Reviewer 3: 用中文的吧

> export AWS_ACCESS_KEY_ID=${access_key}
> export AWS_SECRET_ACCESS_KEY=${secret_key}
> tiup dumpling --host ${host} --port 3306 --user root --password ${password} --filter 'my_db1.table[12]' --no-data --output 's3://my-bucket/schema-backup' --filter "mydb.*"

Author: ```suggestion
export AWS_ACCESS_KEY_ID=${access_key}
export AWS_SECRET_ACCESS_KEY=${secret_key}
tiup dumpling --host ${host} --port 3306 --user root --password ${password} --filter 'my_db1.table[12],mydb.*' --consistency none --no-data --output 's3://my-bucket/schema-backup'
```

#### 1.2 Create the TiDB Lightning configuration file for the schema file

Create a new `tidb-lightning-schema.toml` file, copy the following content into the file, and replace the corresponding content.

```toml
[tidb]

# The target TiDB cluster information.
host = ${host}
port = ${port}
user = "${user_name}
password = "${password}"
status-port = ${status-port}  # The TiDB status port. Usually the port is 10080.
pd-addr = "${ip}:${port}"     # The cluster PD address. Usually the port is 2379.

[tikv-importer]
# "local": Use the default Physical Import Mode (the "local" backend).
# During the import, the target TiDB cluster cannot provide any service.
# For more information about import modes, see https://docs.pingcap.com/tidb/stable/tidb-lightning-overview
backend = "local"

# Set the temporary storage directory for the sorted Key-Value files.
# The directory must be empty, and the storage space must be greater than the size of the dataset to be imported.
# For better import performance, it is recommended to use a directory different from `data-source-dir` and use flash storage,
# which can use I/O exclusively.

> sorted-kv-dir = "/mnt/ssd/sorted-kv-dir"

Reviewer: ```suggestion
sorted-kv-dir = "${path}"
```

#### 2.2 Create the TiDB Lightning configuration file for the data file

Create a new `tidb-lightning-data.toml` configuration file, copy the following content into the file, and replace the corresponding content.

```toml
[tidb]

# The target TiDB cluster information.
host = ${host}
port = ${port}
user = "${user_name}
password = "${password}"
status-port = ${status-port}  # The TiDB status port. Usually the port is 10080.
pd-addr = "${ip}:${port}"     # The cluster PD address. Usually the port is 2379.

[tikv-importer]
# "local": Use the default Physical Import Mode (the "local" backend).
# During the import, the target TiDB cluster cannot provide any service.
# For more information about import modes, see https://docs.pingcap.com/tidb/stable/tidb-lightning-overview
backend = "local"

# Set the temporary storage directory for the sorted Key-Value files.
# The directory must be empty, and the storage space must be greater than the size of the dataset to be imported.
# For better import performance, it is recommended to use a directory different from `data-source-dir` and use flash storage,
# which can use I/O exclusively.

> sorted-kv-dir = "/mnt/ssd/sorted-kv-dir"

Reviewer: ```suggestion
sorted-kv-dir = "${path}"
```

## Prerequisites

> - [Install Dumpling and TiDB Lightning](/migration-tools.md). If you want to create the corresponding tables manually on the target side, don't install Dumpling.

Author: ```suggestion
- [Install Dumpling and TiDB Lightning](/migration-tools.md). If you want to create the corresponding tables manually on the target side, do not install Dumpling.
```

#         During the import, the target TiDB cluster can provide service normally.
backend = "local"

# Set the temporary storage directory for the sorted Key-Value files.
# The directory must be empty, and the storage space must be greater than the size of the dataset to be imported.
# For better import performance, it is recommended to use a directory different from `data-source-dir` and use flash storage,
# which can use I/O exclusively.
sorted-kv-dir = "/mnt/ssd/sorted-kv-dir"

[mydumper]
# The directory of the schema file exported from Amazon Aurora
data-source-dir = "${s3_path}"  # eg: s3://my-bucket/schema-backup
```

If you need to enable TLS in the TiDB cluster, refer to [TiDB Lightning Configuration](/tidb-lightning/tidb-lightning-configuration.md).

#### 1.3 Import the schema file to TiDB

Use TiDB Lightning to import the schema file to the downstream TiDB. If you launch the program directly in the command line, the process might exit unexpectedly after receiving a SIGHUP signal. In this case, it is recommended to run the program using a nohup or screen tool. 

```shell
export AWS_ACCESS_KEY_ID=${access_key}
export AWS_SECRET_ACCESS_KEY=${secret_key}

> tiup tidb-lightning -config tidb-lightning-schema.toml > nohup.out 2>&1 &

Reviewer 2: ```suggestion
nohup tiup tidb-lightning -config tidb-lightning-schema.toml > nohup.out 2>&1 &
```

```

### Step 2. Export and import an Amazon Aurora snapshot to Amazon S3

> This section describes how to export an Amazon Aurora snapshot to Amazon S3.

Reviewer 2: ```suggestion
This section describes how to export an Amazon Aurora snapshot to Amazon S3 and import the snapshot into TiDB by TiDB Lightning.
```

    1 row in set (0.012 sec)
    ```

2. Export the Amazon Aurora snapshot. For detailed steps, refer to [Exporting DB snapshot data to Amazon S3](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/USER_ExportSnapshot.html).

After you obtain the binlog position, export the snapshot within 5 minutes. Otherwise, the recorded binlog position might be outdated and thus cause data conflict during the incremental replication.

After the two steps above, make sure you have the following information ready:

- The Amazon Aurora binlog name and position at the time of the snapshot creation.
- The S3 path where the snapshot is stored, and the SecretKey and AccessKey with access to the S3 path.

### Step 2. Export schema

Because the snapshot file from Amazon Aurora does not contain the DDL statements, you need to export the schema using Dumpling and create the schema in the target database using TiDB Lightning. If you want to manually create the schema, you can skip this step.

> Pass the SecretKey and AccessKey of the account that has access to this Amazon S3 backend storage into the TiDB Lightning node as environment variables. TiDB Lightning also supports reading credential files from `~/.aws/credentials`. This method eliminates the need to pass in the corresponding SecretKey and AccessKey again for all tasks on that TiDB Lightning node.

Author: ```suggestion
Pass the SecretKey and AccessKey of the account that has access to this Amazon S3 storage path into the TiDB Lightning node as environment variables. TiDB Lightning also supports reading credential files from `~/.aws/credentials`. This method eliminates the need to pass in the corresponding SecretKey and AccessKey again for all tasks on that TiDB Lightning node.
```

pd-addr = "${ip}:${port}"     # The cluster PD address, e.g.: 172.16.31.3:2379. TiDB Lightning obtains some information from PD. When backend = "local", you must specify status-port and pd-addr correctly. Otherwise, the import will be abnormal.

[tikv-importer]
# "local": Physical Import Mode. The default backend. The local backend is recommended to import large volumes of data (1 TiB or more). During the import, the target TiDB cluster cannot provide any service.
# "tidb": Logical Import Mode. The "tidb" backend is recommended to import data less than 1 TiB. During the import, the target TiDB cluster can provide service normally.
backend = "local"

# Set the temporary storage directory for the sorted Key-Value files. The directory must be empty, and the storage space must be greater than the size of the dataset to be imported. For better import performance, it is recommended to use a directory different from `data-source-dir` and use flash storage, which can use I/O exclusively.
sorted-kv-dir = "/mnt/ssd/sorted-kv-dir"

[mydump]
# Parse the schema file exported by Dumpling
default-file-rules=true

[mydumper]

> # The path that stores the snapshot file exported from Amazon Aurora.
> data-source-dir = "${s3_path}"  # e.g.: s3://my-bucket/sql-backup

Author: ```suggestion
```

    1 row in set (0.012 sec)
    ```

2. Export the Amazon Aurora snapshot. For detailed steps, refer to [Exporting DB snapshot data to Amazon S3](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/USER_ExportSnapshot.html).

After you obtain the binlog position, export the snapshot within 5 minutes. Otherwise, the recorded binlog position might be outdated and thus cause data conflict during the incremental replication.

After the two steps above, make sure you have the following information ready:

- The Amazon Aurora binlog name and position at the time of the snapshot creation.
- The S3 path where the snapshot is stored, and the secret access key and access key with access to the S3 path.

#### 2.2 Create the TiDB Lightning configuration file for the data file

> Create the `tidb-lightning-data.toml` configuration file as follows. Note that the name of the configuration file here must be different from the name of the configuration file for the schema file.

Reviewer 2: ```suggestion
Create the `tidb-lightning-data.toml` configuration file as follows. Note that the name of the configuration file here must be different from the name of the configuration file for importing the schema file.
```

### Step 4. Import full data to TiDB

> 1. Pass the SecretKey and AccessKey of the account that has access to this Amazon S3 backend storage into the TiDB Lightning node as environment variables. TiDB Lightning also supports reading credential files from `~/.aws/credentials`. When running `tidb-lightning`, if you start the program directly from the command line, it might exit due to the `SIGHUP` signal. It is recommended to use a tool such as `nohup` or `screen`. Refer to the following steps 2 and 3 for details.

Author: ```suggestion
1. Pass the SecretKey and AccessKey of the account that has access to this Amazon S3 storage path into the TiDB Lightning node as environment variables. TiDB Lightning also supports reading credential files from `~/.aws/credentials`. When running `tidb-lightning`, if you start the program directly from the command line, it might exit due to the `SIGHUP` signal. It is recommended to use a tool such as `nohup` or `screen`. Refer to the following steps 2 and 3 for details.
```

2. Use TiDB Lightning to create tables in the downstream TiDB. Skip this step if you have manually created the tables in the target database in advance.

    ```shell
    export AWS_ACCESS_KEY_ID=${access_key}
    export AWS_SECRET_ACCESS_KEY=${secret_key}
    tiup tidb-lightning -config tidb-lightning.toml -d 's3://my-bucket/schema-backup' # Note that the URI here is the one that stores the schema file exported by Dumpling from Amazon Aurora.
    ```

>     For more information about URI configuration parameters, such as specifying an ARN of the AWS IAM role to access S3 data, see [S3 URI Configuration Parameters](/tidb-lightning/tidb-lightning-data-source.md#import-data-from-amazon-s3).

Author: ```suggestion
    For more information about URI configuration parameters, such as specifying an ARN of the AWS IAM role to access S3 data. See [S3 URI Configuration Parameters](/tidb-lightning/tidb-lightning-data-source.md#import-data-from-amazon-s3).
```

> 3. Use TiDB Lightning to import data from Amazon Aurora Snapshot to TiDB.

Author: ```suggestion
3. Use TiDB Lightning to import data from an Amazon Aurora snapshot to TiDB.
```

    ```shell
    export AWS_ACCESS_KEY_ID=${access_key}
    export AWS_SECRET_ACCESS_KEY=${secret_key}

>     nohup tiup tidb-lightning -config tidb-lightning.toml > nohup.out 2>&1 &

Author: ```suggestion
    nohup tiup tidb-lightning -config tidb-lightning.toml -d 's3://my-bucket/sql-backup' > nohup.out 2>&1 &
```
